src/cdk/tree/node.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {FocusableOption} from '@angular/cdk/a11y';
+import {
+  Directive,
+  ElementRef,
+  Input,
+  OnDestroy,
+  TemplateRef
+} from '@angular/core';
+import {takeUntil} from 'rxjs/operator/takeUntil';
+import {Subject} from 'rxjs/Subject';
+import {CdkTree} from './tree';
+import {getTreeControlFunctionsMissingError} from './tree-errors';
+
+
+/** Context provided to the ndoes */
+export class CdkTreeNodeOutletContext<T> {
+
+  static mostRecentContextData: any;
```

Comment:
This is slightly different from what I had in mind. My proposal had been that immediately after calling `createEmbeddedView`, you get the most recent node directive and set the data to it. I think this would be slightly less brittle, since its more reliable than depending on the order of instantiation of the context objects. 

For whatever we end up doing, we should have some very descriptive comments explaining the situation.

Reply:
Now we use a static variable in `CdkTreeNode` and set data to that `mostRecentTreeNode`.